Apps/Database/src/Models/DelegateInvitation.cs

```diff
@@ -0,0 +1,122 @@
+    /// The DelegateInvitation model.
+    /// </summary>
+    public class DelegateInvitation : AuditableEntity
+    {
+        /// <summary>
+        /// Gets or sets the primary key.
+        /// </summary>
+        [Key]
+        [Column("DelegateInvitationId")]
+        public Guid Id { get; set; }
+
+        /// <summary>
+        /// Gets or sets the nickname.
+        /// </summary>
+        [Required]
+        [MaxLength(20)]
+        public string Nickname { get; set; } = null!;
+
+        /// <summary>
+        /// Gets or sets the status for the delegate invitation.
+        /// </summary>
+        [Required]
+        [MaxLength(50)]
+        public DelegateInvitationStatus Status { get; set; }
+
+        /// <summary>
+        /// Gets or sets the sharing code.
+        /// </summary>
+        [Required]
+        [MaxLength(6)]
+        public string SharingCode { get; set; } = null!;
+
+        /// <summary>
+        /// Gets or sets the failed attempts.
+        /// </summary>
+        [Required]
+        public int FailedAttempts { get; set; }
+
+        /// <summary>
+        /// Gets or sets the expiry date.
+        /// </summary>
+        [Required]
+        public DateOnly ExpiryDate { get; set; }
+
+        /// <summary>
+        /// Gets or sets the access for the data sets.
+        /// </summary>
+        [Column(TypeName = "jsonb")]
+        [SuppressMessage("Usage", "CA2227:Collection properties should be read only", Justification = "Team decision")]
+        public HashSet<DataSource>? DataSources { get; set; }
+
+        /// <summary>
+        /// Gets or sets the resource owner hdid.
+        /// </summary>
+        [Required]
+        [MaxLength(52)]
+        public string ResourceOwnerHdid { get; set; } = null!;
+
+        /// <summary>
+        /// Gets or sets the resource owner identifier.
+        /// </summary>
+        [Required]
+        [MaxLength(255)]
+        public string ResourceOwnerIdentifier { get; set; } = null!;
+
+        /// <summary>
+        /// Gets or sets the hdid.
+        /// </summary>
+        [MaxLength(52)]
+        public string? ProfileHdid { get; set; }
+
+        /// <summary>
+        /// Gets or sets the reason code for the delegation.
+        /// </summary>
+        [MaxLength(10)]
+        public ResourceDelegateReason? ReasonCode { get; set; }
+
+        /// <summary>
+        /// Gets or sets a value indicating whether the delegation invitation was removed by the owner.
+        /// </summary>
+        [Required]
+        public bool RemovedByOwner { get; set; }
+
+        /// <summary>
+        /// Gets or sets a value indicating whether the delegation invitation was removed by the delegate.
+        /// </summary>
+        [Required]
+        public bool RemovedByDelegate { get; set; }
+
+        /// <summary>
+        /// Gets or sets the resource delegate.
+        /// </summary>
+        public ResourceDelegate ResourceDelegate { get; set; } = null!;
```

Comment:
```suggestion
        public virtual ResourceDelegate ResourceDelegate { get; set; } = null!;
```
Navigation properties should include the virtual modifier in case we want to enable lazy loading.